### ISSUE TITLE
Ubuntu 17.04 | error while installing gem 'tiny_tds', '~> 1.3.0' | freetds v1.00.21

### ISSUE THREAD

**Issue by shivabhusal** [rails-sqlserver/tiny_tds#379]
> gem 'tiny_tds', '~> 1.3.0'
> ruby v ruby 2.4.0p0
> ruby v ruby 2.3.3p222 

I have Free TDS `v1.00.21` installed on my machine. But whenever I try to install `tiny_tds` I get error
```bash
$ tsql -C

# Compile-time settings (established with the "configure" script)
#                            Version: freetds v1.00.21
#             freetds.conf directory: /usr/local/etc
#     MS db-lib source compatibility: no
#        Sybase binary compatibility: no
#                      Thread safety: yes
#                      iconv library: yes
#                        TDS version: 7.3
#                              iODBC: no
#                           unixodbc: yes
#              SSPI "trusted" logins: no
#                           Kerberos: no
#                            OpenSSL: yes
#                             GnuTLS: no
#                               MARS: no
```

## Error
```ruby
checking for sybdb.h... yes
checking for tdsdbopen() in -lsybdb... yes
checking for dbanydatecrack() in -lsybdb... no
-----
freetds is missing.
Do you have FreeTDS 0.95.80 or higher installed?
-----
*** extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary
libraries and/or headers.  Check the mkmf.log file for more details.  You may
need configuration options.

Provided configuration options:
	--with-opt-dir
	--without-opt-dir
	--with-opt-include
	--without-opt-include=${opt-dir}/include
	--with-opt-lib
	--without-opt-lib=${opt-dir}/lib
	--with-make-prog
	--without-make-prog
	--srcdir=.
	--curdir
	--ruby=/usr/bin/$(RUBY_BASE_NAME)2.3
	--help
	--enable-lookup
	--disable-lookup
	--with-freetds-dir
	--without-freetds-dir
	--with-freetds-include
	--without-freetds-include=${freetds-dir}/include
	--with-freetds-lib
	--without-freetds-lib=${freetds-dir}/lib
	--with-sybdblib
	--without-sybdblib
	--with-sybdblib
	--without-sybdblib
	--enable-system-freetds
	--disable-system-freetds
	--enable-system-iconv
	--disable-system-iconv
	--enable-system-openssl
	--disable-system-openssl
	--enable-gnutls
	--disable-gnutls
	--enable-openssl
	--disable-openssl
	--enable-cross-build
	--disable-cross-build
	--with-freetds-dir
	--without-freetds-dir
	--with-freetds-include
	--without-freetds-include=${freetds-dir}/include
	--with-freetds-lib
	--without-freetds-lib=${freetds-dir}/lib
	--with-sybdblib
	--without-sybdblib
	--with-sybdblib
	--without-sybdblib

To see why this extension failed to compile, please check the mkmf.log which can be found here:

  /var/lib/gems/2.3.0/extensions/x86_64-linux/2.3.0/tiny_tds-1.3.0/mkmf.log

extconf failed, exit code 1

Gem files will remain installed in /var/lib/gems/2.3.0/gems/tiny_tds-1.3.0 for inspection.
Results logged to /var/lib/gems/2.3.0/extensions/x86_64-linux/2.3.0/tiny_tds-1.3.0/gem_make.out

An error occurred while installing tiny_tds (1.3.0), and Bundler cannot continue.
Make sure that `gem install tiny_tds -v '1.3.0'` succeeds before bundling.

In Gemfile:
  activerecord-sqlserver-adapter was resolved to 5.1.1, which depends on
```

**Comment by shivabhusal** [comment]
I successfully installed using `sudo`

```ruby
sudo gem install tiny_tds -v 1.3.0
```

**Comment by faraazahmad** [comment]
@shivabhusal when i run that I get,
```
sudo: gem: command not found
```

**Comment by wilsonsilva** [comment]
You can fix it by installing [FreeTDS](http://www.freetds.org/):

```
apt-get update && apt-get install -yq freetds-bin bash
```

and for MacOS users

```
brew install freetds
```

**Comment by KINGSABRI** [comment]
> You can fix it by installing [FreeTDS](http://www.freetds.org/):
> 
> ```
> apt-get update && apt-get install -yq freetds-bin
> ```
> 
> and for MacOS users
> 
> ```
> brew install freetds
> ```

Thank you for the hint. For me I had to install the dev package
```
apt install freetds-dev
```

**Comment by TB-Development** [comment]
I'm in a similar situation, except my TinyTds version is being resolved to the latest version (2.1.5).  I'm working in a Ruby on Rails application inside of a docker environment that runs on Debian Linux.  My task is to upgrade rails from 3.2.19 to Rails 4.2.

Specs:
* Ruby: 2.3.0
* Rails: 4.2.0
* TinyTDS: 2.1.5
* Debian GNU/Linux 8 (jessie)

I have all of the packages installed listed above with:

```
RUN apt-get install -y --force-yes unixodbc unixodbc-dev freetds-dev freetds-bin tdsodbc
```

Attempting to compile the applications gives me the same error as the OP, as well as installing the gem manually on the container.  The fix is to downgrade to 1.3.0, no other versions upward are working appropriately.  It's weird because there doesn't seem to be any differences between 1.3 and 2.0 in the dependencies.
